docs/core/testing/order-unit-tests.md

```diff
@@ -0,0 +1,88 @@
+
+> [!TIP]
+> In addition to the ordering capabilities outlined in this article, consider [creating custom playlists with Visual Studio](/visualstudio/test/run-unit-tests-with-test-explorer?view=vs-2019#create-custom-playlists) as an alternative.
+
+## [MSTest](#tab/mstest)
+
+### Order alphabetically
+
+With MSTest, tests are automatically ordered by their test name.
+
+> [!NOTE]
+> A test named `Test14` will run before `Test2` even though `2` comes before `14`. This is because, test name ordering does _not_ use natural language sorting.
+
+:::code language="csharp" source="~/dotnet-samples/csharp/unit-testing/MSTest.Project/ByAlphabeticalOrder.cs":::
+
+## [xUnit](#tab/xunit)
+
+The xUnit test framework allows for more granularity and control of test run order. The `ITestCaseOrderer` and `ITestCollectionOrderer` interfaces can be implemented to control the order of test cases for a class, or test collections.
+
+### Order by test case alphabetically
+
+To order test cases by their method name, you could implement the `ITestCaseOrderer` and provide an ordering mechanism.
+
+:::code language="csharp" source="~/dotnet-samples/csharp/unit-testing/XUnit.TestProject/Orderers/AlphabeticalOrderer.cs":::
+
+Then in a test class you would set the test case order with the `TestCaseOrdererAttribute`.
+
+:::code language="csharp" source="~/dotnet-samples/csharp/unit-testing/XUnit.TestProject/ByAlphabeticalOrder.cs":::
+
+### Order by collection alphabetically
+
+To order test collections by their display name, you could implement the `ITestCollectionOrderer` and provide an ordering mechanism.
+
+:::code language="csharp" source="~/dotnet-samples/csharp/unit-testing/XUnit.TestProject/Orderers/DisplayNameOrderer.cs":::
+
+Since test collections have the potential to run in parallel, you need to explicitly disable test parallelization of the collections with the `CollectionBehaviorAttribute`. Then you need to specify the implementation to the `TestCollectionOrdererAttribute`.
+
+:::code language="csharp" source="~/dotnet-samples/csharp/unit-testing/XUnit.TestProject/ByDisplayName.cs":::
+
+### Order by custom attribute
+
+To order xUnit tests with custom attributes, you first need an attribute to rely on. Define a `TestPriorityAttribute` as follows:
+
+:::code language="csharp" source="~/dotnet-samples/csharp/unit-testing/XUnit.TestProject/Attributes/TestPriorityAttribute.cs":::
+
+Next, consider the following `PriorityOrderer` implementation of the `ITestCaseOrderer` interface.
+
+:::code language="csharp" source="~/dotnet-samples/csharp/unit-testing/XUnit.TestProject/Orderers/PriorityOrderer.cs":::
+
+Then in a test class you would set the test case order with the `TestCaseOrdererAttribute` to the `PriorityOrderer`.
+
+:::code language="csharp" source="~/dotnet-samples/csharp/unit-testing/XUnit.TestProject/ByPriorityOrder.cs":::
+
+## [NUnit](#tab/nunit)
+
+### Order alphabetically
+
+With NUnit, tests are automatically ordered by their test name.
+
```

Comment:
Nice article! A correction here - this is actually considered an implementation detail of NUnit, and could change in future. See here: https://github.com/nunit/nunit/issues/1372

Strictly speaking, NUnit has no defined default ordering - the order attribute is the way to enforce it if needed. 🙂

Reply:
Thank you very much @ChrisMaddock - this information is very helpful. If you and the NUnit team decide to add this functionality later or, I'd be happy to collaborate with you on adding it. 🙏